.pre-commit-config.yaml

```diff
@@ -6,7 +6,7 @@ repos:
         language_version: python3
 
   - repo: https://github.com/PyCQA/flake8
-    rev: 4.0.1
+    rev: 7.0.0
```

Comment:
let's add this to the CHANGELOG.md under 
**UNRELEASED -> For developers of the library**

Reply:
Should I update other pre-commit hooks ? That way we are up to date and developers will not have to `pre-commit install` anytime soon.

Reply:
I updated all pre-commit hooks and change the argument for pyupgrade. See changelog (or commit description) for more info